packages/astro/src/vite-plugin-build-html/index.ts

```diff
@@ -138,6 +140,10 @@ export function rollupPluginAstroBuildHTML(options: PluginOptions): VitePlugin {
               const src = getAttribute(node, 'src');
               if (src?.startsWith(srcRoot) && !astroAssetMap.has(src)) {
                 astroAssetMap.set(src, fs.readFile(new URL(`file://${src}`)));
+              } else if (src?.startsWith(srcRootWeb) && !astroAssetMap.has(src)) {
+                const resolved = new URL('.' + src, astroConfig.projectRoot);
+                const assetId = viteifyURL(resolved);
```

Comment:
This may need a different function than `viteifyURL`. Because this is Rollup, this may need `slash(fileURLToPath(resolved))` which _should_ match the Rollup ID 🤞🏻 (feel free to make this a method).

Reply:
To add a little more context, there are 2 types of IDs according to Vite:

1. `url`: `/@fs/…`
2. `id`: `/Users/astro…` (Linux) or `C:/Users/…` (Windows)

The former can be used for the module graph and in the dev server, but not Rollup. The latter can be used for the module graph and Rollup, but not the dev server.

#2052 changed `viteifyURL` to be `url` (“dev“). Should we have some distinction here between “dev” IDs and “build” IDs? Overall this resolution gets a little confusing and would love feedback on how to make it safer / more predictable.

Reply:
Resolving for simplicity